baker/archival/ArchivalUtils.ts

```diff
@@ -0,0 +1,71 @@
+import { AssetMap, dayjs, lazy, omit } from "@ourworldindata/utils"
+import {
+    GrapherChecksums,
+    GrapherChecksumsObjectWithHash,
+} from "./ArchivalChecksumUtils.js"
+import { simpleGit } from "simple-git"
+
+export interface ArchivalTimestamp {
+    date: Date
+    formattedDate: string
+}
+
+export interface ArchivalManifest {
+    assets: {
+        static: AssetMap
+        runtime: AssetMap
+    }
+    archivalDate: string
+    chartId: number
+    chartConfigId: string
+    chartSlug: string
+    checksums: GrapherChecksums
+    checksumsHashed: string
+    commitShas: {
+        "owid-grapher"?: string
+    }
+}
+
+const DATE_TIME_FORMAT = "YYYYMMDD-HHmmss"
+
+export const getDateForArchival = (): ArchivalTimestamp => {
+    const date = dayjs().utc()
+    const formattedDate = date.format(DATE_TIME_FORMAT)
+
+    return { date: date.toDate(), formattedDate }
+}
+
+const getOwidGrapherCommitSha = lazy(async () => {
+    if (process.env.BUILDKITE_COMMIT) return process.env.BUILDKITE_COMMIT
+    try {
+        return await simpleGit().revparse(["HEAD"])
+    } catch {
+        return undefined
+    }
+})
+
+export const assembleManifest = async (manifestInfo: {
```

Comment:
You can see an example of a manifest here: http://staging-site-archived-chart-versions:3000/20250218-211057/grapher/access-to-basic-services.manifest.json (behind Tailscale)